### PR TITLE
Adds conditional rendering for top level link and refactors Mobile and Desktop components into discreet renderers

### PR DESCRIPTION
**Pull request by rosschapman**
## Description 

- [x] Adds support for a top level link
- [x] Adds separate rendering containers for Mobile and Desktop views
- [x] Discreetly separates styling for Mobile and Desktop views

### CODE REVIEW

**Review by adriencyberspace — CHANGES_REQUESTED** [review]
I'm currently blocked on being able to create a 2-tiered navigation but a couple of things I'm seeing right now:

1. If you have two dropdown menus, they both open no matter which top level link you click. I think only the selected menu should open.
2. Also a style issue is that the links don't extend the full width of the menu but should, which is noticeable on hover or focus

image

File: app/components/ui/Navigation.tsx
Comment (adriencyberspace) [reply app/components/ui/Navigation.tsx:71]: Works for me. You could also use a `switch` but not a big deal with only 3 cases.

**Comment by adriencyberspace** [comment]
I think we should hide mobile on desktop size screens. i.e. if I open it on a smaller screen, it should disappear when I make my screen larger.

image

**Comment by rosschapman** [comment]
@adriencyberspace my latest [commit](https://github.com/Exygy/askdarcel-web/pull/75/commits/818c85dd462e9554d52dddeee9bc74f91f7106cb) fixes the issues with multiple _Navigation Menu_ types in the nav. Regarding  https://github.com/Exygy/askdarcel-web/pull/75#issuecomment-2215471398 I propose we consider that in a polish phase. I'd like to better understand the scenario of a user resizing their browser that much. 

Please take another look thank you!

https://github.com/Exygy/askdarcel-web/assets/5185/79c017b1-94bc-483e-b05a-adf50f13665f

**Comment by adriencyberspace** [comment]
@rosschapman Off the bat, I'm seeing that the top level and single level links do not have uniform styling but do in the mockup. 

Actual:
image

Mockup:
image

Aside from that, can you make the mobile absolute positioned? I know it came in from SF Service Finder, but I can't recall seeing a mobile nav that pushes content over on any other site and I think it affects UX.

**Comment by adriencyberspace** [comment]
I also still think these links should be full container width:

image

**Comment by rosschapman** [comment]
@adriencyberspace These are fixed with https://github.com/Exygy/askdarcel-web/pull/75/commits/7d5efd2cb8b08af49d4b386eadaa9204f5150751
- [ ] https://github.com/Exygy/askdarcel-web/pull/75#issuecomment-2218366138
- [ ] https://github.com/Exygy/askdarcel-web/pull/75#issuecomment-2218366138

Regarding doing a slide-out vs push-out nav...hmmm. That will be a bit more involved. Let's park that for another iteration. I'll need to get some more design guidance on that to figure out how the overlay control looks/works.

**Review by adriencyberspace — APPROVED** [review]
LGTM! Two minor visual QA changes before merging:

1. Single tier item on mobile is not same font weight as two tiered
image
2. Single tier item on desktop does not have same hover color as two tiered

**Pull request merged**
